File: orderer.cpp
#include"orderer.h"
#include<iostream>
int main(){
	int s, o=0;
	cout<<"Size: ";
	cin>>s;
	vector<int> r;
	while(o<factorial(s)){
		r=getorder(o, s);
		for(int i=0; i<s; i++){
			cout<<r[i];
		}
		cout<<endl;
		o++;
	}
	return 0;
}

File: orderer.py
import math

def getorder(order, l):
    orig=[0]*l
    facts=[0]*l
    c=0
    while c<l:
        orig[c]=c
        facts[c]=math.factorial(c)
        c+=1
    out=[]
    takefrom=orig
    #print("I'll take from", takefrom)
    while len(out)<l:
        divider=facts[l-len(out)-1]
        #print("Dividing bY:", divider)
        totake=int(order/divider)
        order=order%divider
        #print("REST:", order)
        out.append(takefrom[totake])
        takefrom.pop(totake)
        #print("TOKEN:", out[len(out)-1])
    return out

while True:
    s=int(input("Size: "))
    o=0
    while o<math.factorial(s):
        print("Result:", getorder(o, s))
        o+=1
    
    
